common/persistence/visibility/store/standard/cassandra/visibility_store.go

```diff
@@ -460,8 +437,47 @@ func (v *visibilityStore) ListClosedWorkflowExecutionsByStatus(
 	return response, nil
 }
 
-// DeleteWorkflowExecution is a no-op since deletes are auto-handled by cassandra TTLs
-func (v *visibilityStore) DeleteWorkflowExecution(_ *manager.VisibilityDeleteWorkflowExecutionRequest) error {
+func (v *visibilityStore) DeleteWorkflowExecution(request *manager.VisibilityDeleteWorkflowExecutionRequest) error {
+	// Primary key in closed_execution table has close_time which is not in the request.
+	// Query closed_execution table first (using secondary index by workflow_id) to get close_time.
+	// This is not very efficient if workflow has multiple executions.
```

Comment:
I don't like this part, this is one extra read for every visibility delete. Did you try to put close_time into task?

Reply:
Changed.